conf/root.go

```diff
@@ -101,5 +134,77 @@ var (
 func init() {
 	createContextCmd.Flags().StringVarP(&confFileTemplate, "templatefile", "f", "", "-f <config file to build context from>")
 	dumpCmd.Flags().StringVarP(&dumpOutfile, "outfile", "o", "", "-o <outfile")
-	RootCmd.AddCommand(dumpCmd, createContextCmd, getContextCmd, getContextsCmd, useContextCmd, deleteContextCmd)
+	RootCmd.AddCommand(dumpCmd, createContextCmd, currentContextCmd, getContextsCmd, useContextCmd, deleteContextCmd, setCmd, editCmd)
+}
+
+func setConfigWithViper(key string, value string) error {
+	v := viper.New()
+	cname, err := GetCurrentContextName()
+	if err != nil {
+		return err
+	}
+	basedir, err := getContextBaseDir()
+	if err != nil {
+		return err
+	}
+	v.SetConfigFile(filepath.Join(basedir, cname))
+	v.SetConfigType("toml")
+	if err = v.ReadInConfig(); err != nil {
+		return err
+	}
+	if !v.IsSet(key) {
+		return fmt.Errorf("Key '%s' not found in config.", key)
+	}
```

Comment:
this will block us from setting a new KV pair though right? would we want this ability?

Reply:
Since contexts are saved by serializing the struct, all values should already exist (even if just initialized to their defaults). For complex config changes like adding more nodes, just use `audius-d config edit`.

But you're right that this may be needed in the future, I just think we can cross that bridge when we get to it.